Sources/StreamChat/Utils/MessagesPaginationStateHandling/MessagesPaginationStateHandling.swift

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A component responsible for handling the messages pagination state.
+protocol MessagesPaginationStateHandling {
+    /// The current state of the messages pagination.
+    var state: MessagesPaginationState { get }
+
+    /// This function should be called before performing the pagination call.
+    func start(pagination: MessagesPagination)
```

Comment:
I believe `begin` is a better match than `start` as it actually means `to initiate an action`.